File: features/ccmoon/F5_ST.py
import time
def gen_f5_st(f5_st):
   #unix時間を取得
   f5_st_part = f5_st.split("z")[3]
   unix_time = f5_st_part
   admIdleTimeout = 3600
   admGuardTime = 100
   sessionstart = unix_time
   maxTimeout = 604800
   maxGuardTime = 600
   
   F5_ST=f"{unix_time}c{admIdleTimeout}c{admGuardTime}c{sessionstart}c{maxTimeout}c{maxGuardTime}c"
   return F5_ST

File: lib/features/ccmoon/f5_st.dart
/// F5 BIG-IP の `F5_ST` cookie を長寿命版へ作り変える。
///
/// 元 cookie のフォーマットが想定外なら例外を投げる
/// (Python版は `split("z")[3]` で IndexError になっていた)。
String genF5St(String f5st) {
  final parts = f5st.split('z');
  if (parts.length < 4) {
    throw FormatException('F5_ST cookie の形式が想定外です: $f5st');
  }
  final unixTime = parts[3];
  const admIdleTimeout = 3600;
  const admGuardTime = 100;
  const maxTimeout = 604800;
  const maxGuardTime = 600;
  return '${unixTime}c${admIdleTimeout}c${admGuardTime}c'
      '${unixTime}c${maxTimeout}c${maxGuardTime}c';
}
